### DISCUSSION TITLE
safely-migrating-passwords-in-asp-net-core-identity-with-a-custom-passwordhasher/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#117] [Announcements]
# safely-migrating-passwords-in-asp-net-core-identity-with-a-custom-passwordhasher/

In this post I create a custom IPasswordHasher<> implementation that can be used in ASP.NET Core Identity to migrate from weak hashes like MD5 or SHA1.

https://andrewlock.net/safely-migrating-passwords-in-asp-net-core-identity-with-a-custom-passwordhasher/

**Comment by stevenmcneese** [comment]
I have an old application that stored 96 byte salt and hashed password from the concatenation of the input password and generated salt using the Crypto.HashPassword method of the System.Web.Helper.Crypto library in .net 4.6.  The app is being rewritten in .Net 7.  I am trying to figure out how to recreate the Crypto.VerifyHashedPassword currently being used and ultimately regenerate the password using the new method in ASP .Net Core Identity.  I am struggling trying to find away to verify the input password using the old hash algorithm.  Do you have any suggestions?

**Reply by andrewlock** [reply]
[This stack overflow question](https://stackoverflow.com/questions/61411373/replace-system-web-helpers-crypto-in-net-core) shows the approach I would take: 

> create your own Crypto type and copy source code from here: https://github.com/aspnet/AspNetWebStack/blob/master/src/System.Web.Helpers/Crypto.cs

[a comment by an automated account is not shown]
